sympy/core/tests/test_numbers.py

```diff
@@ -1749,3 +1748,43 @@ def test_Integer_precision():
     assert Float('1.0', precision=Integer(15))._prec == 15
     assert type(Float('1.0', precision=Integer(15))._prec) == int
     assert sympify(srepr(Float('1.0', precision=15))) == Float('1.0', precision=15)
+
+def test_numpy():
+    from sympy.external import import_module
+    from sympy.utilities.pytest import skip
+    np = import_module('numpy')
+    
+    if not np:
+        skip('numpy not installed. Abort numpy tests.')
+
+    def equal(x, y):
+        return x == y and type(x) == type(y)
+    print(type(S(123)))
+    print(type(Float(np.int_(123))))
+    try:
+        assert equal(
+            Float(np.int_(12345678912345678)), S(12345678912345678.))
+        assert equal(
+            Float(np.intp(12345678912345678)), S(12345678912345678.))
+    except OverflowError:
+        # May fail on 32-bit systems: Python int too large to convert to C long
+        pass
+    assert equal(Float(np.intc(1234567891)), S(1234567891.))
+    assert equal(Float(np.int8(-123)), S(-123.))
+    assert equal(Float(np.int16(-12345)), S(-12345.))
+    assert equal(Float(np.int32(-1234567891)), S(-1234567891.))
+    assert equal(
+        Float(np.int64(-12345678912345678)), S(-12345678912345678.))
+    assert equal(Float(np.uint8(123)), S(123.))
+    assert equal(Float(np.uint16(12345)), S(12345.))
+    assert equal(Float(np.uint32(1234567891)), S(1234567891.))
+    assert equal(
+        Float(np.uint64(1234567891234567891)), S(1234567891234567891.))
+    assert equal(Float(np.float32(1.123456)), Float(1.123456, precision=24))
+    assert equal(Float(np.float64(1.1234567891234)),
+                 Float(1.1234567891234, precision=53))
+    assert equal(Float(np.longdouble(1.123456789)),
+                 Float(1.123456789, precision=80))
```

Comment:
Is longdouble always precision 80? Could it be quad precision?

Reply:
I believe it is always 80 - the documentation states that longdouble is padded with zeros to either 96 or 128 bits.
Source: https://docs.scipy.org/doc/numpy-dev/user/basics.types.html

Reply:
I think it is architecture dependent.

Reply:
@cbm755 you are right - the long double comes from C and is general depends on architecture and compiler. Don't know an easy solution to test for different architectures so removed it along with `np.longcomplex` from test_numbers.py for now.